packages/components/Menu/src/Menu/useMenu.ts

```diff
@@ -61,6 +54,13 @@ const useMenuOpenState = (
 
   const state = isControlled ? open : openInternal;
 
+  // Due to how events get fired we get double notifications
+  // for the same event causing us to immediately reopen
+  // a menu when we close it. Adding in a delay to prevent
+  // this behavior.
+  const delayOpen = 150;
+  const [lastCloseTimestamp, setLastCloseTimestamp] = React.useState<number>(-1);
```

Comment:
I think using a ref would be better - state causes a new render cycle, and the state wouldn't update to the new value until that render cycle is complete. This value isn't needed directly to render anything i.e. changing it shouldn't cause any renders

Reply:
That makes sense! Making the changes and posting it in a new PR that's based off the main branch